src/org/abymaly/code/magnets/Magnets.java

```diff
@@ -2,4 +2,158 @@
+                } else {
+                    sbLine.append(" ");
+                    sbLine.append(board[i][j]);
+                    sbLine.append(" ");
+                }
+            }
+            sbLine.append(newLine);
+        }
+        System.out.println(sbLine);
+    }
+
+    public void setStartPosition(Signs[][] board, int x, int y) {
+        board[x][y] = Signs.O;
+    }
+
+    public void setData(Signs[][] board, int x, int y, Signs value) {
+        board[x][y] = value;
+    }
+
+    public Signs[][] emptyBoard () {
+        Signs [][] newBoard = new Signs[size][size];
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                setData(newBoard, i, j, Signs.EMPTY);
+            }
+        }
+        return newBoard;
+    }
+
+    public void refreshStartPosition(Signs[][] board) {
+        setStartPosition(board, startPosition.x, startPosition.y);
+    }
+
+    public void emptyStartPosition (Signs[][] board) {
+        setData(board, startPosition.x, startPosition.y, Signs.EMPTY);
+    }
+
+    public void getStartPosition(Signs[][] board) {
+
+        for (int i = 0; i < size; ++i) {
+            for (int j = 0; j < size; ++j) {
+                if (board[i][j].equals(Signs.O)) {
+                    startPosition.x = i;
+                    startPosition.y = j;
+                }
+            }
+        }
+    }
+
+    public boolean moveTop(Signs[][] board) {
+        if (startPosition.x == 0 || board[(startPosition.x) - 1][startPosition.y].equals(Signs.X)) {
+            return false;
+        } else {
+            emptyStartPosition(board);
+            ++possibleMoves;
+            --startPosition.x;
+            refreshStartPosition(board);
+        }
+        return true;
+    }
+
+    public boolean moveRight(Signs[][] board) {
+        if ((startPosition.y == (size - 1)) || board[startPosition.x][(startPosition.y) + 1].equals(Signs.X)) {
+            return false;
+        } else {
+            emptyStartPosition(board);
+            ++possibleMoves;
+            ++startPosition.y;
+            refreshStartPosition(board);
+        }
+        return true;
+    }
+
+    public boolean moveBottom(Signs[][] board) {
+        if ((startPosition.x == size - 1) || board[startPosition.x + 1][startPosition.y].equals(Signs.X)) {
+            return false;
+        } else {
+            emptyStartPosition(board);
+            ++possibleMoves;
+            ++startPosition.x;
+            refreshStartPosition(board);
+        }
+        return true;
+    }
+
+    public boolean moveLeft(Signs[][] board) {
+        if ((startPosition.y == 0) || board[startPosition.x][startPosition.y - 1].equals(Signs.X)) {
+            return false;
+        } else {
+            emptyStartPosition(board);
+            ++possibleMoves;
+            --startPosition.y;
+            refreshStartPosition(board);
+        }
+        return true;
+    }
+
+    public void go(Signs[][] board, int[] moves) {
```

Comment:
A czemu te `moves[]` to jest tablica `int`ów, a nie jakichś `enum`ów???
Jak zobaczyłem to w teście, to nie miałem pojęcia co która liczba znaczy.

A gdyby był enum, to każda wartość mogłaby mieć metody "ruchWPoziomie' i `ruchWPionie` i w ogóle by nie trzaeb było robić switcha. Ten motyw był ćwiczony na lekcji z kółkiem i krzyżykiem. 

----

Miło by było, gdyby go zwracało jakaś liczbę